code/datums/abilities/critter/stomach_retreat.dm

```diff
@@ -0,0 +1,76 @@
+/datum/targetable/critter/stomach_retreat
+	name = "Retreat to Stomach"
+	desc = "Turn yourself inside out for shelter! Must be inside a disposal chute."
+	icon_state = "mimic_stomach_retreat"
+	cooldown = 120 SECONDS
+	cooldown_after_action = TRUE
+	needs_turf = FALSE
+	var/inside = FALSE
+	var/obj/current_container = null
+	var/last_appearance = null
+
+
+	New()
+		..()
+		if (!holder.owner.GetComponent(/datum/component/death_barf))
+			holder.owner.AddComponent(/datum/component/death_barf)
+
+	cast(atom/target)
+		. = ..()
+		if (inside)
+			switch(tgui_alert(holder.owner, "Leave yourself?", "Retreat to Stomach", list("Yes.", "No.")))
+				if ("Yes.")
+					deactivate()
+					return TRUE
+				if ("No.")
+					return TRUE
+		var/obj/target_container = holder.owner.loc
+		if (!isobj(target_container))
+			boutput(holder.owner, SPAN_ALERT("There isn't anything to climb into here!"))
```

Comment:
Kind of misleading, should indicate that you must be already inside something to cast the ability